arm9/source/utils/scripting.c

```diff
@@ -1846,18 +2074,20 @@ bool MemToCViewer(const char* text, u32 len, const char* title) {
 bool FileTextViewer(const char* path, bool as_script) {
     // load text file (completely into memory)
     // text file needs to fit inside the STD_BUFFER_SIZE
-    u32 flen, len;
+    size_t fileSize = FileGetSize(path);
+    if (fileSize >= STD_BUFFER_SIZE) {
```

Comment:
Max file size check. I tested with a much lower number as well to make sure that the editor would not allow characters to be typed beyond the max file size.